### PR TITLE
llama fp8 - enable non reuse cache flow for fp8 (#64)

### PR DESCRIPTION
**Pull request by HolyFalafel**
* llama fp8 - enable non reuse cache flow for fp8

remove depracted kv cache fp8 flag

Change-Id: Id76f94a127dee202376e8f27de7b28f58affedae

* fixing lm eval

Change-Id: I230fa53e7b49d8bb36397b063f652ba3def84600

* remove old quantization mode

Change-Id: I538172f29870311349ed79d928cfacc60fb534e8

### CODE REVIEW

File: examples/text-generation/utils.py [hunk 1]
```diff
@@ -373,15 +373,22 @@ def initialize_model(args, logger):
...
         else setup_distributed_model(args, model_dtype, model_kwargs, logger)
     )
     tokenizer, model = setup_tokenizer(args, model)
     generation_config = setup_generation_config(args, model, tokenizer)
+
+    if args.const_serialization_path:
+        setup_const_serialization(args.const_serialization_path)
     if args.fp8:
-        model = setup_quantization(args, model)
+        model = setup_inference(args, model)
```
Comment (libinta) [inline examples/text-generation/utils.py:391]: why can't we directly use htcore.hpu_initialize(model) ?

Comment (HolyFalafel) [reply examples/text-generation/utils.py:391]: We want to do it organized with prints and all

Comment (libinta) [reply examples/text-generation/utils.py:391]: also the setup_inference is confusing as this util is for text-generation inference. can you move the import print and htcore.hpu_initialize(model) here?

**Review by libinta — APPROVED** [review]

**Comment by regisss** [comment]
@HolyFalafel There is some overlap between this PR and #780, I see the same changes for `--const_serialization_path`

**Comment by HolyFalafel** [comment]
> @HolyFalafel There is some overlap between this PR and #780, I see the same changes for `--const_serialization_path`

Yes, in order to avoid merge conflicts, I had to base this commit on top of the commit which is relevant to #780.
So we should merge the other one first.

**Comment by szutenberg** [comment]
Hi @HolyFalafel ,

I see that "Code check quality" tests are failing. Could you fix formatting with ruff?
Any ETA to have these changes merged?

**Comment by regisss** [comment]
> Hi @HolyFalafel ,
> 
> I see that "Code check quality" tests are failing. Could you fix formatting with ruff? Any ETA to have these changes merged?

We need to merge https://github.com/huggingface/optimum-habana/pull/780 first. I'll try it out with Synapse 1.14 but not sure it will work. If it does I'll merge it, ortherwise we'll need to wait for the release of Synapse 1.15.

File: optimum/habana/transformers/generation/configuration_utils.py [hunk 2]
```diff
@@ -27,10 +27,6 @@ class GaudiGenerationConfig(GenerationConfig):
         If negative (default=-1) pad to max if `static_shapes` is set. Else start with
         `shape = bucket_size * ceil(prompt_len/bucket_size)` and then grow space by `bucket_size` when needed.
         Only active if `static_shapes` is used. Can't be used with `reuse_cache`.
-    bucket_internal (`bool`, *optional*):
-        Split kv sequence into buckets in decode phase. It improves throughput when max_new_tokens is large.
```
Comment (libinta) [inline optimum/habana/transformers/generation/configuration_utils.py:31]: why do you remove the bucket_internal, can you add it back?

Comment (HolyFalafel) [reply optimum/habana/transformers/generation/configuration_utils.py:31]: Not sure, the cherry-pick ended without errors. I'll readd it

**Review by libinta — APPROVED** [review]

**Comment by regisss** [comment]
@HolyFalafel We'll merge this PR in the `synapse_1.15` branch but there are merge conflicts. I think it's due to the upgrade to Transformers 4.38, can you check please?

**Review by regisss — APPROVED** [review]

**Pull request merged**